charts/cluster-addons/templates/monitoring/loki.yaml

```diff
@@ -0,0 +1,123 @@
+{{- if and .Values.monitoring.enabled .Values.monitoring.loki_v2.enabled }}
+# TODO: Update this to include Loki v3 once the new addon is available
+{{- if and .Values.monitoring.lokiStack.enabled .Values.monitoring.loki_v2.enabled }}
+{{- fail (printf "Cannot enable both 'lokiStack' and 'loki' addons simultaneously; please choose one.") }}
+{{- end }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "loki") }}-config
+  labels:
+    {{- include "cluster-addons.componentLabels" (list . "loki") | nindent 4 }}
+    addons.stackhpc.com/watch: ""
+stringData:
+  defaults: |
+    serviceMonitor:
+      enabled: true
+    monitoring:
+      dashboards:
+        enabled: false
+  overrides: |
+    {{- toYaml .Values.monitoring.loki_v2.release.values | nindent 4 }}
+---
+apiVersion: addons.stackhpc.com/v1alpha1
+kind: HelmRelease
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "loki") }}
+  labels: {{ include "cluster-addons.componentLabels" (list . "loki") | nindent 4 }}
+spec:
+  clusterName: {{ include "cluster-addons.clusterName" . }}
+  bootstrap: true
+  chart: {{ toYaml .Values.monitoring.loki_v2.chart | nindent 4 }}
+  targetNamespace: {{ .Values.monitoring.loki_v2.release.namespace }}
+  releaseName: loki
+  valuesSources:
+    - secret:
+        name: {{ include "cluster-addons.componentName" (list . "loki") }}-config
+        key: defaults
+    - secret:
+        name: {{ include "cluster-addons.componentName" (list . "loki") }}-config
+        key: overrides
+  # StatefulSets do not allow their PVCs to be resized, but we can use lifecycle hooks
+  # to ensure that it happens correctly
+  lifecycleHooks:
+    preUpgrade:
+      - apiVersion: apps/v1
+        kind: StatefulSet
+        name: loki
+        action: delete
+    {{- if .Values.monitoring.loki_v2.release.values.singleBinary.persistence.enabled }}
+      - apiVersion: v1
+        kind: PersistentVolumeClaim
+        selector:
+          app: loki
+          release: loki-stack
+        action: patch
+        options:
+          data:
+            spec:
+              resources:
+                requests:
+                  storage: {{ .Values.monitoring.loki_v2.release.values.singleBinary.persistence.size }}
+    {{- else }}
+    postUpgrade:
+      - apiVersion: v1
+        kind: PersistentVolumeClaim
+        selector:
+          app: loki
+          release: loki-stack
+        action: delete
+    {{- end }}
```

Comment:
Can you test that these hooks still work with the new chart please? I think the process should be:
1. Deploy cluster with loki_v2 enabled
2. Increase the persistence.size config value then run a `helm upgrade`
3. Check that the PVC and corresponding Cinder volume get correctly resized
4. Set persistence.enabled = false and run another Helm upgrade
5. Check that the PVC and corresponding Cinder volume get deleted
6. Delete the test cluster